pipeline/builder.go

```diff
@@ -37,11 +38,18 @@ func build(c pipelineBuilderOptions) (Pipeline, error) {
 	}
 
 	switch {
-	case c.batchDecoder != nil:
-		setupBatchProcessor(c, p)
-	case c.decoder != nil:
-		setupProcessor(c, p)
+	case c.messagePool != nil:
+		err = setupMessagePool(c, p)
+	case len(c.inputStreams) > 0:
+		err = setupStreamProcessor(c, p)
+	default:
+		err = errors.E(op, "invalid pipeline configuration")
 	}
+
+	if err != nil {
+		return nil, err
```

Comment:
Nit: seems like some switch branches will add `op` and others not. You could add `op` here instead of in the `default` flow